File: 2337-remove-digit-from-number-to-maximize-result/remove-digit-from-number-to-maximize-result.java
class Solution {
    public String removeDigit(String number, char digit) {
        String ans="";
        for(int i=0;i<number.length();i++)
        {
            if(number.charAt(i)==digit)
            {
                if(ans.length()==0)
                ans=number.substring(0,i)+number.substring(i+1);
                else
                {
                    String s=number.substring(0,i)+number.substring(i+1);
                    if(ans.compareTo(s)<0)
                    {
                        ans=s;
                    }
                }
            }
        }
        return ans;
    }
}

File: 2337-remove-digit-from-number-to-maximize-result/remove-digit-from-number-to-maximize-result.cpp
class Solution {
public:
    string removeDigit(string number, char digit) {
        string ans = "";
        for (int i = 0; i < number.length(); i++) {
            if (number[i] == digit) {
                if (ans.length() == 0)
                    ans = number.substr(0, i) + number.substr(i + 1);
                else {
                    string s = number.substr(0, i) + number.substr(i + 1);
                    if (ans.compare(s) < 0) {
                        ans = s;
                    }
                }
            }
        }
        return ans;
    }
};
